doc/source/data/api/grouped_dataset.rst

```diff
@@ -3,29 +3,54 @@
 GroupedDataset API
 ==================
 
-.. autoclass:: ray.data.grouped_dataset.GroupedDataset
-    :members:
+.. currentmodule:: ray.data
```

Comment:
This page is not only refactored, but also restructured.

Reference: 
Pandas groupby - https://pandas.pydata.org/docs/reference/groupby.html
PySpark groupby - https://spark.apache.org/docs/latest/api/python/reference/pyspark.pandas/groupby.html